lib/matplotlib/cbook.py

```diff
@@ -1012,20 +1012,18 @@ def mkdirs(newdir, mode=0o777):
         > mkdir -p NEWDIR
         > chmod MODE NEWDIR
     """
+    # this functionality is now in core python as of 3.2
+    if six.PY3:
+        os.makedirs(newdir, mode=mode, exist_ok=True)
+        return
+    # LPY DROP
+    # the rest of this function can be removed when we drop legacy python
     try:
-        if not os.path.exists(newdir):
-            parts = os.path.split(newdir)
-            for i in range(1, len(parts) + 1):
-                thispart = os.path.join(*parts[:i])
-                if not os.path.exists(thispart):
-                    os.makedirs(thispart, mode)
-
-    except OSError as err:
-        # Reraise the error unless it's about an already existing directory
-        if err.errno != errno.EEXIST or not os.path.isdir(newdir):
+        os.makedirs(newdir, mode=mode)
```

Comment:
Do we know that this works with multiple processes? I thought the whole point of `exist_ok` in Python3 was to handle the case of the directory pre-existing, and I thought the initial attempt at rewriting the code to look at each leaf separately was to work around the multi-process problem. Isn't this code now back to how it was before the change to protect from multiple processes attempting to create the directory? My previous comment was that `os.makedirs` looked like it should just be `os.mkdir` as the code was only ever creating a directory at a time.